### PR TITLE
feat: check update profile

### PR DESCRIPTION
**Pull request by leeduy1**

### CODE REVIEW

File: src/components/modules/SidebarSettings/SidebarSettings.tsx [hunk 1]
```diff
@@ -61,6 +58,63 @@ const SidebarSettings: React.FC = () => {
...
+      initUserInfo.phone !== newInfoUser.phone ||
+      initUserInfo.description !== newInfoUser.description
+    ) {
+      setDisabled(false);
+    } else {
+      setDisabled(true);
+    }
+  }, [newInfoUser]);
+
+  const handleFormChange = (
```
Comment (vuidv54) [inline src/components/modules/SidebarSettings/SidebarSettings.tsx]: handleValueChange

File: src/components/modules/SidebarSettings/SidebarSettings.tsx [hunk 2]
```diff
@@ -61,6 +51,68 @@ const SidebarSettings: React.FC = () => {
...
+    return {
+      fullName: userProfileStore?.fullName,
+      location: userProfileStore?.location,
+      email: userProfileStore?.email,
+      phone: userProfileStore?.phone,
+      description: userProfileStore?.description,
+    };
+  }, [updateSuccess]);
+
+  const handleValueChage = (
```
Comment (henefisa) [inline src/components/modules/SidebarSettings/SidebarSettings.tsx]: sai chinh ta

File: src/components/modules/SidebarSettings/SidebarSettings.tsx [hunk 3]
```diff
@@ -61,6 +51,68 @@ const SidebarSettings: React.FC = () => {
...
+    e: {
+      fullName: string;
+      location: string;
+      email: string;
+      phone: string;
+      description: string;
+    },
+    newInfoUser: TUserInfo
```
Comment (henefisa) [inline src/components/modules/SidebarSettings/SidebarSettings.tsx]: viet ra interfaces

File: src/components/modules/SidebarSettings/SidebarSettings.tsx [hunk 4]
```diff
@@ -61,6 +51,68 @@ const SidebarSettings: React.FC = () => {
...
+      initUserInfo.fullName !== newUserInfo.fullName ||
+      initUserInfo.location !== newUserInfo.location ||
+      initUserInfo.phone !== newUserInfo.phone ||
+      initUserInfo.description !== newUserInfo.description
+    ) {
+      setDisabled(false);
+    } else {
+      setDisabled(true);
+    }
+  };
```
Comment (henefisa) [inline src/components/modules/SidebarSettings/SidebarSettings.tsx]: viet util de check

**Review by henefisa — CHANGES_REQUESTED** [review]

File: src/components/modules/SidebarSettings/SidebarSettings.tsx [hunk 5]
```diff
@@ -61,6 +52,39 @@ const SidebarSettings: React.FC = () => {
...
+
+  const handleFinish = async (values: TUserInfo) => {
+    if (selectedFile) {
+      const res = await userService.getPresignUrl(selectedFile.name, t);
+      await userService.putPresignUrl(res.url, selectedFile, t);
+      const user = { ...values, avatar: res.key };
+      await userService.updateUserProfile(dispatch, user, t);
+    } else {
+      await userService.updateUserProfile(dispatch, values, t);
+    }
```
Comment (henefisa) [inline src/components/modules/SidebarSettings/SidebarSettings.tsx]: refactor

File: src/models/user.ts [hunk 6]
```diff
@@ -53,6 +53,12 @@ export type TUserInfo = Omit<
   'id' | 'created_at' | 'updated_at' | 'status' | 'role'
 >;
 
+export interface IChangeField {
+  fullName: null | string | undefined;
+  phone: null | string | undefined;
+  description: null | string | undefined;
+  location: null | string | undefined;
```
Comment (henefisa) [inline src/models/user.ts]: refactor

File: src/components/modules/SidebarSettings/SidebarSettings.tsx [hunk 7]
```diff
@@ -61,6 +52,39 @@ const SidebarSettings: React.FC = () => {
...
+  const initUserInfo = React.useMemo(() => {
+    return {
+      fullName: userProfileStore?.fullName,
+      location: userProfileStore?.location,
+      phone: userProfileStore?.phone,
+      description: userProfileStore?.description,
+    };
+  }, [updateSuccess]);
+
+  const handleValueChange = (e: IChangeField, newInfoUser: TUserInfo) => {
```
Comment (henefisa) [inline src/components/modules/SidebarSettings/SidebarSettings.tsx]: handleValueChange -> handleValuesChange

File: src/utils/checkdisabled.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    initUserInfo.fullName !== newUserInfo.fullName ||
+    initUserInfo.location !== newUserInfo.location ||
+    initUserInfo.phone !== newUserInfo.phone ||
+    initUserInfo.description !== newUserInfo.description
+  ) {
+    setDisabled(false);
+  } else {
+    setDisabled(true);
+  }
+};
```
Comment (henefisa) [inline src/utils/checkdisabled.ts]: refactor

File: src/utils/checkdisabled.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import { SetStateAction } from 'react';
+import { IChangeField } from 'src/models';
+
```
Comment (henefisa) [inline src/utils/checkdisabled.ts]: src/utils/checkdisabled.ts -> sai convention dat ten

**Review by henefisa — CHANGES_REQUESTED** [review]

File: src/components/modules/SidebarSettings/SidebarSettings.tsx [hunk 10]
```diff
@@ -61,6 +52,39 @@ const SidebarSettings: React.FC = () => {
...
+      description: e.description || newInfoUser.description,
+    };
+    compareObjects(initUserInfo, newUserInfo, setDisabled);
+  };
+
+  const handleFinish = async (values: TUserInfo) => {
+    if (selectedFile) {
+      const res = await userService.getPresignUrl(selectedFile.name, t);
+      await userService.putPresignUrl(res.url, selectedFile, t);
+      const user = { ...values, avatar: res.key };
```
Comment (henefisa) [inline src/components/modules/SidebarSettings/SidebarSettings.tsx]: user -> profile

File: src/utils/compareObjects.ts [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+import { IChangeField } from 'src/models/user';
+import isEqual from 'lodash.isequal';
+
+export const compareObjects = (
+  initUserInfo: IChangeField,
+  newUserInfo: IChangeField,
+  setDisabled: { (value: SetStateAction<boolean>): void; (arg0: boolean): void }
+) => {
+  isEqual(initUserInfo, newUserInfo) ? setDisabled(true) : setDisabled(false);
+};
```
Comment (henefisa) [inline src/utils/compareObjects.ts]: remove

Comment (henefisa) [reply src/utils/compareObjects.ts]: compare thẳng trong handler

**Review by henefisa — CHANGES_REQUESTED** [review]

**Review by henefisa — APPROVED** [review]

**Pull request merged**